### PR TITLE
Add basic automated tests

### PR DESCRIPTION
**Pull request by brikis98**
This PR adds a very basic level of automated tests to the Gruntwork installer. It runs a bare-minimum-end-to-end integration test: run the bootstrap script to install `gruntwork-install` and then run `gruntwork-install` to install a module from script-modules. 

The tests run in a Docker container to keep your local environment clean.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -143,4 +156,4 @@ that'll happen when you execute it is a harmless syntax error.
 
 1. Add support for a `--version` flag to `bootstrap-gruntwork-installer.sh` and `gruntwork-install`.
 1. Configure a CI build to automatically set the `--version` flag for each release.
-1. Add automated tests for this repo.
+1. Add an `uninstall` command that uses an `uninstall.sh` script in each module.
```
Comment (josh-padnick) [inline README.md]: An interesting idea. But do we really need it? Wouldn't we be more likely to build a new AMI / Docker image without the installed script module?

File: gruntwork-install [hunk 2]
```diff
@@ -88,9 +88,11 @@ function fetch_script_module {
 function validate_module {
   local readonly module_name="$1"
   local readonly download_path="$2"
+  local readonly tag="$3"
+  local readonly branch="$4"
 
   if [[ ! -e "$download_path/$module_name" ]]; then
-    echo "ERROR: No files were downloaded. Are you sure \"$module_name\" is a valid Script Module in $SCRIPT_MODULES_REPO?"
+    echo "ERROR: No files were downloaded. Are you sure \"$module_name\" is a valid Script Module in $SCRIPT_MODULES_REPO (tag = $tag, branch = $branch)?"
```
Comment (josh-padnick) [inline gruntwork-install]: Nice.

File: test/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,9 @@
+FROM ubuntu
```
Comment (josh-padnick) [inline test/Dockerfile]: You may want to use a tag other than the implicit `latest` for stability reasons here.

File: test/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+installer:
+  image: gruntwork/gruntwork-installer
+  volumes:
+    # Mount the scripts in the root directory under /src
+    - ../:/src
+    # Mount the test code in this directory under /test
+    - .:/test
```
Comment (josh-padnick) [inline test/docker-compose.yml]: Wouldn't these two mounts overlap each other? Is that actually an issue?

**Comment by josh-padnick** [comment]
Great to see automated tests here. Feel free to merge once you've responded to my comments.

File: README.md (hunk 1 above)
Comment (brikis98) [inline README.md]: I suspect people will install some of our scripts on their dev boxes, at which point, uninstall becomes important.

File: test/Dockerfile (hunk 3 above)
Comment (brikis98) [inline test/Dockerfile]: Fixed

File: test/docker-compose.yml (hunk 4 above)
Comment (brikis98) [inline test/docker-compose.yml]: I suppose `/src` will contain the test code too. A bit messy, but not a big concern.

**Comment by brikis98** [comment]
Merging...

**Pull request merged**